ground/src/main/java/com/google/android/ground/model/submission/SkippedTaskData.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.model.submission
+
+import kotlinx.serialization.Serializable
+
+/** User skipped the response for the given task. */
+@Serializable
+class SkippedTaskData : TaskData {
```

Comment:
This isn't a new task type, so why create a new `TaskData` type? Could we instead just add a `skipped` flag to `TaskData` base class?

Reply:
I had started with this approach but faced couple of challenges:

1. It was possible to have both skipped and value set at the same time. This led to adding more complexity in each task type fragment / view model to ensure skipped and value are never used simultaneously.

2. Serialization of task data became more complicated as now we have to store one extra bit in each task type.

Both of these problems were solved by creating a new task type which works like a empty value but with a special intent (user skipped).

Please let me know if you feel otherwise regarding the currently chosen approach.

Reply:
I don't really have a sense for why serialization would be more complete - you could always check the `skipped` field first and short circuit if set. As for (1), it almost sounds as if you're tricking the UI into resetting its state by setting a different task data.. this sounds somewhat magical and likely adds complexity.

That said, if you've tried this already and it's expensive to implement, I defer to your judgement.

Reply:
Can you please LG if the changes look ok to you?